### DISCUSSION TITLE
Support ctrl-w in the Snacks input

### DISCUSSION THREAD

**Discussion by mortymacs** [folke/snacks.nvim#376] [Q&A] [answered]
Hi,
When I try to remove a word in Snacks, similar to how LSP Rename works for renaming a variable, it just changes the focus to a different window. I'm not sure if this is an issue with my configuration or the input module.
So, how to enable text-related keys in the input?
My input config: https://github.com/mortymacs/dotfiles/blob/main/home-manager/common/nvim/lua/lua/misc.lua#L52-L57

**Comment by dpetka2001** [comment, marked as answer]
This is what worked for me 
```lua
input = {
  win = {
    keys = {
      i_del_word = { "<C-w>", "delete_word", mode = "i" },
    },
    actions = {
      delete_word = function()
        return "<cmd>normal! diw<cr><right>"
      end,
    },
  },
},
```

**Reply by mortymacs** [reply]
Awesome! thanks!

**Reply by hellovietduc** [reply]
I'm trying to map this to Option + Backspace on MacOS but it doesn't work. Do you happen to know why?

```lua
    input = {
      win = {
        keys = {
          i_del_word = { "<M-BS>", "delete_word", mode = "i" },
        },
        actions = {
          delete_word = function()
            return "<cmd>normal! diw<cr><right>"
          end,
        },
      },
    },
```

**Reply by dpetka2001** [reply]
Try adding `i_del_word = { "<M-BS>", "delete_word", mode = "i", expr = true }`.

**Reply by hellovietduc** [reply]
I tried that but it didn't work. I wonder if it's because Snacks input can't recognize the Option key in MacOS as `M`. I have a keymap for `<M-BS>` in normal buffer insert mode and it works.

**Reply by Geobert** [reply]
I’m trying similar thing with `["<c-BS>"] = { "<c-w>", mode = {"i"}, expr = true }` with no luck

**Comment by mklishevych** [comment]
In case someone else arrives here with the same request and the proposed solution doesn't work, here is what worked for me:
```lua
input = {
  enabled = true,
  win = {
    border = "single",
    -- https://github.com/folke/snacks.nvim/discussions/376
    keys = {
      i_del_word = { "<C-w>", "delete_word", mode = "i" },
    },
    actions = {
      delete_word = function()
        vim.cmd("normal! diw<cr>")
      end,
    },
  },
},
```

**Reply by dpetka2001** [reply]
Or add `expr = true` where you define the mapping with my previous answer. But both are equivalent.
